python/cudf_polars/tests/expressions/test_rolling.py

```diff
@@ -19,21 +25,106 @@ def test_rolling():
     ]
     df = (
         pl.DataFrame({"dt": dates, "a": [3, 7, 5, 9, 2, 1]})
-        .with_columns(pl.col("dt").str.strptime(pl.Datetime))
+        .with_columns(pl.col("dt").str.strptime(pl.Datetime(time_unit)))
         .lazy()
     )
     q = df.with_columns(
         sum_a=pl.sum("a").rolling(index_column="dt", period="2d"),
-        min_a=pl.min("a").rolling(index_column="dt", period="2d"),
-        max_a=pl.max("a").rolling(index_column="dt", period="2d"),
+        min_a=pl.min("a").rolling(index_column="dt", period="5d"),
+        max_a=pl.max("a").rolling(index_column="dt", period="10d", offset="2d"),
+    )
+
+    assert_gpu_result_equal(q)
+
+
+def test_rolling_date_raises():
+    dates = [
+        "2020-01-01",
+        "2020-01-01",
+        "2020-01-01",
+        "2020-01-02",
+        "2020-01-03",
+        "2020-01-08",
+    ]
+    df = (
+        pl.DataFrame({"dt": dates, "a": [3, 7, 5, 9, 2, 1]})
+        .with_columns(pl.col("dt").str.strptime(pl.Date()))
+        .lazy()
+    )
+    q = df.with_columns(
+        max_a=pl.max("a").rolling(index_column="dt", period="10d", offset="2d"),
     )
 
     assert_ir_translation_raises(q, NotImplementedError)
 
 
+@pytest.mark.parametrize("dtype", [pl.Int32, pl.UInt32, pl.Int64, pl.UInt64])
+def test_rolling_integral_orderby(dtype):
+    df = pl.LazyFrame(
+        {
+            "orderby": pl.Series([1, 4, 8, 10, 12, 13, 14, 22], dtype=dtype),
+            "values": [1, 2, 3, 4, 5, 6, 7, 8],
+        }
+    )
+    q = df.with_columns(
+        pl.col("values").sum().rolling("orderby", period="4i", closed="both")
+    )
+
+    assert_gpu_result_equal(q)
+
+
+def test_rolling_collect_list_raises():
+    df = pl.LazyFrame(
+        {
+            "orderby": [1, 4, 8, 10, 12, 13, 14, 22],
+            "values": [1, 2, 3, 4, 5, 6, 7, 8],
+        }
+    )
+    assert_ir_translation_raises(
+        df.with_columns(pl.col("values").rolling("orderby", period="4i")),
+        NotImplementedError,
+    )
+
+
+def test_unsorted_raises_computeerror():
+    df = pl.LazyFrame({"orderby": [1, 2, 4, 2], "values": [1, 2, 3, 4]})
+    q = df.select(pl.col("values").sum().rolling("orderby", period="2i"))
+    with pytest.raises(pl.exceptions.InvalidOperationError):
+        q.collect(engine="in-memory")
```

Comment:
I'm trying to decide if we want this assertion. IIUC we're asserting some behavior about polars; nothing about cudf-polars. I guess if that changes in polars, we'll want this test to fail so that we know we should fix it?

If that's the motivation, then this seems fine.

Reply:
Exactly that.